lib/geordi/gitlinear.rb

```diff
@@ -0,0 +1,201 @@
+
+      if branch_name.present?
+        checkout_branch branch_name, new_branch: false
+      else
+        checkout_branch new_branch_name, new_branch: true, from_master: from_master
+      end
+    end
+
+    private
+
+    attr_accessor :highline, :settings
+
+    def choose_issue
+      loading_message = 'Connecting to Linear ...'
+      print(loading_message)
+      issues = applicable_issues
+      reset_loading_message = "\r#{' ' * (loading_message.length + issues.length)}\r"
+
+      Geordi::Interaction.fail('No issues to offer.') if issues.empty?
+
+      if Util.testing?
+        return issues[0]
+      end
+
+      highline.choose do |menu|
+        menu.header = 'Choose an issue'
+
+        issues.each do |issue|
+          state = issue['state']['name']
+          if issue['assignee']
+            assignee = issue['assignee']['name']
+            assignee_is_me = issue['assignee']['isMe']
+          else
+            assignee = "none"
+            assignee_is_me = false
+          end
+
+          if state == 'In Progress'
+            state = HighLine::GREEN + state + HighLine::RESET
+          else
+            state = HighLine::RED + state + HighLine::RESET
+          end
+
+          state += HighLine::BOLD if assignee_is_me
+
+          label = "(#{assignee}, #{state}) #{issue['title']}"
+          label = bold(label) if assignee_is_me
+
+          menu.choice(label) { return issue }
+        end
+
+        menu.hidden ''
+        print reset_loading_message # Once menu is build
+      end
+
+      nil # Return nothing
+    end
+
+    def create_commit(title, description, *git_args)
+      extra = highline.ask("\nAdd an optional message").strip
+      title << ' - ' << extra if extra != ''
+      Util.run!(['git', 'commit', '--allow-empty', '-m', title, '-m', description, *git_args])
+    end
+
+    def applicable_issues
+      if Util.testing?
+        settings.linear_api_key
+        return ENV['GEORDI_TESTING_NO_LINEAR_ISSUES'] == 'true' ? [] : [{
+                                                                          'identifier' => '12',
+                                                                          'title' => 'Test Issue',
+                                                                          'url' => 'https://www.issue-url.com',
+                                                                          'branchName' => 'testuser/12-test-issue'
+                                                                        }]
+      end
+
+      team_ids = settings.linear_team_ids
+      filter = {
+        "team": {
+          "id": {
+            "in": team_ids,
+          }
+        },
+        "state": {
+          "type": {
+            "eq": "started"
+          }
+        }
+      }
+      response = request_with_payload(<<~GRAPHQL, filter: filter)
+        query Issues($filter: IssueFilter) {
+          issues(filter: $filter) {
```

Comment:
I have no experience in using GraphQL. Are all these "filters" necessary?

Reply:
I will have a look at this at Monday and come back to you about it. I do not have time to research this at this moment. Maybe there is a shorter solution

Reply:
I did a bit of research on the topic by now and I do not think there is any better way to do this. This is the recommended way for inserting variabes by [GraphQL](https://graphql.org/learn/queries/#variables).

Alternatively, you could insert the filter directly into the query like it is described in the [Linear API](https://developers.linear.app/docs/graphql/working-with-the-graphql-api/filtering). But since our filter is a bit longer and more complex than the examples, I feel like the query gets more readable if we keep the filter as a seperate variable.

Reply:
Ok, fine then.